cmd/nimbess-cni.go

```diff
@@ -164,7 +202,54 @@ func cmdAdd(args *skel.CmdArgs) error {
 		CNIVersion: conf.CNIVersion,
 	}
 
-	// TODO: handle IPAM type and data
+	// Run the IPAM plugin and get back the config to apply
+	ipamRequest, err := ipam.ExecAdd(cniRequest.IpamType, args.StdinData)
+	if err != nil {
+		return err
+	}
+
+	// Invoke ipam del if err to avoid ip leak
+	defer func() {
+		if err != nil {
+			ipam.ExecDel(cniRequest.IpamType, args.StdinData)
+		}
+	}()
+
+	// Convert whatever the IPAM result was into the current Result type
```

Comment:
I think this should say: Convert IPAM result received to the correct format for the result version